indico/web/client/js/react/components/WTFSearchDropdown.jsx

```diff
@@ -144,10 +151,14 @@ export function RemoteSearchDropdown({
   );
 
   useEffect(() => {
+    if (optionsFromProps.length !== 0) {
+      setOptions(transformOptions(optionsFromProps));
+      return;
+    }
     if (preload) {
       fetchData();
     }
-  }, [preload, fetchData]);
+  }, [preload, fetchData, optionsFromProps, transformOptions]);
```

Comment:
FYI this only works because the component is used in a kind of special way, and `optionsFromProps` is always the same object (not just the same data in the object). Otherwise this would cause a render loop, since e.g. `{} != {}`. But since this is always populated from a statically defined object, it works fine.

Reply:
oh okay, got it! Good to understand more the intricacies of JS